engine_test.go

```diff
@@ -319,6 +319,50 @@ func TestDDL(t *testing.T) {
 	require.Equal(s, testTable.Schema())
 }
 
+func TestNaturalJoin(t *testing.T) {
```

Comment:
We should test some more cases:

1. NATURAL JOIN on two tables with no common column at all.
2. NATURAL JOIN on two tables with a common column, some rows match the join condition, but some others do not.